src/dpg_slam/dpg_slam.cc

```diff
@@ -748,6 +899,31 @@ namespace dpg_slam {
         return cell_loc;
     }
 
+    void occupancyGrid::combineOccupancyGrids(const occupancyGrid &grid1, const occupancyGrid &grid2) {
+   		 
+   	gridInfo = grid1.gridInfo;	
+    	occupied_cell_info_ = grid1.occupied_cell_info_;
+	std::unordered_map<cellKey, OccupiedCellInfo, boost::hash<cellKey>> occupied_cell_info2 = grid2.occupied_cell_info_;
+	
+	// Insert elements of second grid appropriately.
+	for (const auto &occ_grid_entry : grid2.gridInfo) {
+		cellKey key = occ_grid_entry.first;
+		CellStatus value = occ_grid_entry.second;
+		 OccupiedCellInfo cellInfoValue = occupied_cell_info2[key];
+		// Returns false if key is common so we can handle it seperately.
+		bool isNewCell = gridInfo.emplace(key, value).second;
+		occupied_cell_info_.emplace(key, cellInfoValue);
+		// Handle common cells by prioritizing OCCUPIED
+		if (!isNewCell)	{
+			if (value == OCCUPIED){
+				gridInfo[key] = value;
+				occupied_cell_info_[key] = cellInfoValue;
```

Comment:
I think you need to append to occupied_cell_info_[key] rather than replace if both values are occupied. 

Approximately 
```
std::vector<PointIdInfo> points = occupied_cell_info_[key].points_;
points.insert(points.end(), cellInfoValue.points_.begin(), cellInfoValue.points_.end())
occupied_cell_info_[key].points_ = points;
```

Reply:
you're right. My bad.